builtin_array.go

```diff
@@ -786,6 +786,7 @@ func (r *Runtime) arrayproto_map(call FunctionCall) Value {
 	if _, stdSrc := o.self.(*arrayObject); stdSrc {
 		if arr, ok := a.self.(*arrayObject); ok {
 			values := make([]Value, length)
```

Comment:
A concern I had about this approach was the potential for double counting memory, I'm including a breakdown about why this is not an issue.

Reply:
`values` variable memory utilization goes untracked.